skopt/learning/gaussian_process/kernels.py

```diff
@@ -287,12 +287,13 @@ def gradient_x(self, x, X_train):
             self.k2.gradient_x(x, X_train)
         )
 
+
 class Product(Kernel, sk_Product):
 
     def gradient_x(self, x, X_train):
         x = np.asarray(x)
+        x = np.expand_dims(x, axis=0)
         X_train = np.asarray(X_train)
-        X = np.expand_dims(x, axis=0)
```

Comment:
is there a test we can add that would have detected this? To stop the :beetle: :ghost: from coming back